sql/core/src/main/scala/org/apache/spark/sql/sources/ddl.scala

```diff
@@ -83,10 +99,104 @@ private[sql] class DDLParser extends StandardTokenParsers with PackratParsers wi
   protected lazy val className: Parser[String] = repsep(ident, ".") ^^ { case s => s.mkString(".")}
 
   protected lazy val pair: Parser[(String, String)] = ident ~ stringLit ^^ { case k ~ v => (k,v) }
+
+  protected lazy val column: Parser[StructField] =
+  ( ident ~  ident ^^ { case name ~ typ =>
+      StructField(name, metastoreTypes.toDataType(typ))
+    }
+  |
+    ident ~ (DECIMAL ~ "(" ~> numericLit) ~ ("," ~> numericLit <~ ")") ^^ {
+      case name ~ precision ~ scale =>
+        StructField(name, DecimalType(precision.toInt, scale.toInt))
+    }
+  )
+}
+
+/**
+ * :: DeveloperApi ::
+ * Provides a parser for data types.
+ */
+@DeveloperApi
+private[sql] class MetastoreTypes extends RegexParsers {
+  protected lazy val primitiveType: Parser[DataType] =
+    "string" ^^^ StringType |
```

Comment:
We'd better consider the lowercase and upper case of `type` which user defined.
e.g:
```
name STRING,  
    salary FLOAT,  
    subordinates ARRAY<STRING>,  
    deductions MAP<STRING, FLOAT>,  
    address STRUCT<street:STRING, city:STRING, state:STRING, zip:INT>  
```